provider/core/src/serde.rs

```diff
@@ -25,81 +25,156 @@ use std::ops::Deref;
 use std::rc::Rc;
 use yoke::*;
 
-/// An object that receives data from a Serde Deserializer. Implemented by [`DataPayload`].
+/// A wrapper around a type `T` implementing `serde::Deserialize`.
 ///
-/// Lifetimes:
+/// Workaround for [compiler bug #85636](https://github.com/rust-lang/rust/issues/85636).
+#[repr(transparent)]
+#[derive(serde::Deserialize)]
+pub struct SerdeDeDataStructWrap<T>(pub T);
```

Comment:
I added the wrapper struct close to where I needed it.  I could instead put it in the `yoke` crate.  Thoughts?

Reply:
I would just call it YokeTraitHack or something to make it clear that it's a hack. We could stick it in the yoke crate

Reply:
Done.